File: tessoku_book/a42.cpp
#include <iostream>
#include <vector>
#include<algorithm>
using namespace std;
typedef long long ll;
#define rep0(i,n) for(int i=0;i<=n;i++)
#define rep(i,n) for(int i=1;i<=n;i++)
int n,k,a[309],b[309];

int getScore(int c,int d){
    int cnt=0;
    rep(i,n){
        if(c<=a[i]<=c+k && d<=b[i]<=d+k)cnt+=1;
    }
    return cnt;
}

int main(){
    int answer=0;
    cin>>n>>k;
    rep(i,n)cin>>a[i]>>b[i];
    rep(i,100)rep(j,100){
        int Score=getScore(i,j);
        answer=max(answer,Score);
    }
    cout<<answer<<endl;
    return 0;
}

File: tessoku_book/a42.py
def count(i,j,k,a,b):
    cnt = 0
    for u in range(n):
        if i <= a[u] and a[u] <= i+k and j<=b[u] and b[u] <= k+j:
            cnt += 1
    return cnt
    
n,k = map(int, input().split())

a = [0 for _ in range(n)]
b = [0 for _ in range(n)]

for i in range(n):
    a[i],b[i] = map(int, input().split())

score = 0
for i in range(1,101):
    for j in range(1, 101):
        score = max(score, count(i,j,a,b,k))
        